### ISSUE TITLE
Segment remainder of Pali texts

### ISSUE THREAD

**Issue by sujato** [suttacentral/suttacentral#920]
## Texts to segment

**Mostly done, see 2-do below ⇩**

We still have to segment the remainder of the Pali texts. So far what has been completed is:

- Vinaya, 
- 4 nikayas, 
- dhp, thag and thig

So these are not part of this issue. However, two texts of thag were ommitted from bilara, namel thag17.2 and thag21.1, and these have been added to these files. See #1503 

The aim is to make consistent and semantic segments, while removing all HTML and other things that are not part of the text itself.

### Khuddaka

1. <s>Dhammapada (dhp)</s>
2. Udana (ud)
3. Itivuttaka (iti)
4. Sutta Nipata (snp)
5. <s>Theragatha (thag)</s>
6. <s>Therigatha (thig)</s>
1. Bv
1. vv
1. Cnd
1. Mnd
1. Cp
1. Ja
1. Kp
1. Mil
1. Ne
1. Pe
1. Ps
1. Pv
1. tha-ap
1. thi-ap

### Vinaya

1. Parivara

### Abhidhamma

1. ds
1. dt
1. kv
1. patthana
1. pp
1. vb
1. ya

## Files

Here are the files, in form they are identical with the Vinaya files that were used for conversion. Note they include a minimal CSS sheet if you want to inspect them visually.

[pli_remainder_segmented2.zip](https://github.com/suttacentral/suttacentral/files/3689273/pli_remainder_segmented2.zip)

## To-do

- [x] Re-generate the segment numbers
- [x] Convert to JSON for bilara.
- [x] Ensure that no information has been lost. In particular, check the metadata source of the HTML files. In some cases (eg. Bi Pc 91) there is an editorial comment. Thus not all metadata fields can be assumed as identical.

**Comment by sujato** [comment]
For now, let us wait until Bilara is ready.

**Comment by sujato** [comment]
Closed by mistake, see 2-do for remaining issues.

**Comment by ayya-vimala** [comment]
@Sujato ... could you run me through what needs to be done here exactly? I need a little bit more info for doing this as I have not followed it. The sooner these things are done the better so I can also get a good and finished dataset for Buddhanexus to work with on the next run. As Pali is only the EBTs at the moment and not all extra commentaries, as well as that it is well-segmented, it is fairly easy for the neural network to work with; it is probably the best coded-dataset we have.

**Comment by sujato** [comment]
I don't think there's any need for you to do anything ATM. Blake will ready this for Bilara, as he did with the Vinaya. He's already got the system down; it is somewhat complex, especially generating the segment numbers. Once this is done, you can pull all the Pali texts from bilara-data, they will finally be in a consistent and clear segmented state!

Ultimately it would be nice to apply the segmenting principles to the commentary, so you can match any Pali text with its commentary :boom:

**Comment by ayya-vimala** [comment]
Any good digital source for the commentarial texts you can recommend?

**Comment by sujato** [comment]
As for the conversion, you'd have to ask Blake, I'm not sure how automated it is. But it's not really valuable in general: it just does the specific job of converting these kind of html files (which use a highly weird markup) to bilara.

As for the commentaries, the VRI tipitaka is a good source, perhaps the only digitized source. usefully, it is based on mainly the same numbering system as the Mahasangiti. And the spelling conventions and so on are usually the same, too. Of course, the HTML source is not as well structured.

**Comment by ayya-vimala** [comment]
Actually, VRI has a nicely structured XML source. Looks very do-able. Will need to look into multiprocessing over 4 cores though! This is pretty huge.
